1.2/percentile_test.go

```diff
@@ -61,3 +59,11 @@ func TestGenerateInvalidInput(t *testing.T) {
 
 	}
 }
+
+func Must(fn func(float32) (percentile, error), x float32) percentile {
+	v, err := fn(x)
+	if err != nil {
+		log.Fatalln(err)
```

Comment:
The must pattern should panic.  Here is why:

What is `v` when `err != nil`?  `v`is probably `nil` and if you return `nil` from a Must fuction, the calling code is going to be sad and will probably panic... except now the panic is further from the root cause.  Try it: What happens if you declare a function variable, and then invoke it without ever setting it?  E.g.

```
type foo func() int
func main() {
    var f foo // initialized to nil
    println(f())
}
```

I'd refactor the signature to look like `func mustGeneratePercentile(x float32) percentile`, which then internally calls `generatePercentile` and panics if that errors.  This has a nice parallel to the `mustCompile` I linked earlier.

Must allows callers to skip error handling (at the risk of a panic) by saying "I know the input to this function satisfies its requirements".  Useful in test code where you know that `generatePercentile(.05)` won't error.  It allows you to use your initial pattern where the rendered function was part of the tests struct instead of being generated on the fly in the test.

It is entirely fine to skip the `must` pattern if you don't need it.  The general go philosophy is to pass errors as values and force your callers to deal with them.  Folks won't be upset if you do that, whereas they'll be rather upset if code panics when it could have errored.

Reply:
I thought I should avoid `panic` at all costs, but makes sense here. Updated.